gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/JsonElementConversionFactory.java

```diff
@@ -0,0 +1,505 @@
+import parquet.example.data.simple.BooleanValue;
+import parquet.example.data.simple.DoubleValue;
+import parquet.example.data.simple.FloatValue;
+import parquet.example.data.simple.IntegerValue;
+import parquet.example.data.simple.LongValue;
+import parquet.io.api.Binary;
+import parquet.schema.GroupType;
+import parquet.schema.MessageType;
+import parquet.schema.PrimitiveType;
+import parquet.schema.PrimitiveType.PrimitiveTypeName;
+import parquet.schema.Type;
+import parquet.schema.Types;
+
+import static org.apache.gobblin.converter.parquet.JsonElementConversionFactory.RecordConverter.RecordType.CHILD;
+import static org.apache.gobblin.converter.parquet.JsonSchema.DataType.*;
+import static parquet.schema.OriginalType.UTF8;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.BINARY;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.INT32;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.INT64;
+import static parquet.schema.Type.Repetition.REPEATED;
+
+
+/**
+ * <p>
+ * Creates a JsonElement to Parquet converter for all supported data types.
+ * </p>
+ *
+ * @author tilakpatidar
+ *
+ */
+public class JsonElementConversionFactory {
+
+  /**
+   * Use to create a converter for a single field from a parquetSchema.
+   *
+   * @param schemaNode
+   * @param state
+   * @param repeated - Is the {@link Type} repeated in the parent {@link Group}
+   * @return
+   */
+  public static JsonElementConverter getConverter(JsonSchema schemaNode, WorkUnitState state, boolean repeated) {
+
+    DataType fieldType = schemaNode.getType();
+    switch (fieldType) {
+      case INT:
+        return new IntConverter(schemaNode, repeated);
+
+      case LONG:
+        return new LongConverter(schemaNode, repeated);
+
+      case FLOAT:
+        return new FloatConverter(schemaNode, repeated);
+
+      case DOUBLE:
+        return new DoubleConverter(schemaNode, repeated);
+
+      case BOOLEAN:
+        return new BooleanConverter(schemaNode, repeated);
+
+      case STRING:
+        return new StringConverter(schemaNode, repeated);
+
+      case ARRAY:
+        return new ArrayConverter(schemaNode, state);
+
+      case ENUM:
+        return new EnumConverter(schemaNode, state);
+
+      case RECORD:
+        return new RecordConverter(schemaNode, state);
+
+      case MAP:
+        return new MapConverter(schemaNode, state);
+
+      default:
+        throw new UnsupportedOperationException(fieldType + " is unsupported");
+    }
+  }
+
+  /**
+   * Converts a JsonElement into a supported ParquetType
+   * @author tilakpatidar
+   *
+   */
+  public static abstract class JsonElementConverter {
+    protected final boolean repeated;
+    protected final JsonSchema jsonSchema;
+    private PrimitiveTypeName targetType;
+
+    /**
+     * HashMap to convert {@link DataType} to {@link PrimitiveTypeName}
+     */
+    private final static HashMap<DataType, PrimitiveTypeName> typeMap = new HashMap<>();
```

Comment:
Is this used?

Reply:
Not anymore. Forgot to remove it after refactoring.